pipeline-steps/common.groovy

```diff
@@ -208,4 +208,44 @@ def gen_instance_name(){
   return instance_name
 }
 
+def archive_artifacts(){
+  try{
+    sh """#!/bin/bash
+    d="artifacts_\${BUILD_TAG}"
+    mkdir -p \$d
+
+    # logs and config from the single use slave
+    mkdir -p \$d/\$HOSTNAME/log
+    cp -rp /openstack/log/\$HOSTNAME-* \$d/\$HOSTNAME/log ||:
+    cp -rp /etc/ \$d/\$HOSTNAME/etc
+    cp -rp /var/log/ \$d/\$HOSTNAME/var_log
+
+    # logs and config from the containers
+    while read c
+    do
+      mkdir -p \$d/\$c/log
+      cp -rp /openstack/log/\$c/* \$d/\$c/log ||:
+      cp -rp /var/lib/lxc/\$c/rootfs/etc \$d/\$c ||:
+    done < <(lxc-ls)
+
+    # compress to reduce storage space requirements
+    tar cjf "\$d".tar.bz2 \$d
+
+    # create an index so people can see whats available without downloading
+    # and unarchiving the tar.
+    find \$d > artifact_index.txt
+    """
+  } catch (e){
+    print(e)
+    throw(e)
+  } finally{
+    // still worth trying to archiveArtifacts even if some part of
+    // artifact collection failed.
+    archiveArtifacts(
+      allowEmptyArchive: true,
+      artifacts: 'artifacts_*.tar.bz2,artifact_index.txt'
```

Comment:
Do we want to archive all the artifacts here? This would include logs from old test runs too, right?